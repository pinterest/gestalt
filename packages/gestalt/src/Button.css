:root {
  --blue: #4a90e2;
  --darkGray: #333;
  --red: #e60023;
  --lightGray: #efefef;
  --white: #fff;
}

.button {
  composes: borderBox from "./Layout.css";
}

.solid {
  composes: noBorder from "./Borders.css";
}

.enabled {
  composes: accessibilityOutline from "./Focus.css";
  composes: pointer from "./Cursor.css";
}

.sm {
  composes: small from "./Layout.css";
  padding: 8px 14px;
}

.md {
  composes: medium from "./Layout.css";
  padding: 10px 14px;
}

.lg {
  composes: large from "./Layout.css";
  padding: 12px 14px;
}

.block {
  composes: xsCol12 from "./Column.css";
  composes: block from "./Layout.css";
}

.inline {
  composes: inlineBlock from "./Layout.css";
}

.disabled {
  composes: lightGrayBg from "./Colors.css";
  cursor: default;
}

.gray {
  composes: lightGrayBg from "./Colors.css";
}

.gray:hover,
.gray:focus {
  background-color: #e2e2e2;
}

.gray:active {
  background-color: #dadada;
}

.red {
  composes: redBg from "./Colors.css";
}

.red:hover,
.red:focus {
  background-color: #ad081b;
}

.red:active {
  background-color: #a3081a;
}

.blue {
  composes: blueBg from "./Colors.css";
}

.blue:hover,
.blue:focus {
  background-color: #4a8ad4;
}

.blue:active {
  background-color: #4a85c9;
}

.transparent {
  composes: transparentBg from "./Colors.css";
  border: 1px solid #fff;
}

.transparent:hover,
.transparent:focus {
  background-color: rgba(0, 0, 0, 0.06);
}

.transparent:active {
  background-color: rgba(0, 0, 0, 0.1);
}

.white {
  composes: whiteBg from "./Colors.css";
}

.white:hover,
.white:focus {
  background-color: #f1f1f1;
}

.white:active {
  background-color: #e8e8e8;
}
