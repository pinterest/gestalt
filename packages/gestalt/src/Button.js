// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import borderStyles from './Borders.css';
import styles from './Button.css';
import Text from './Text.js';

type Props = {|
  accessibilityExpanded?: boolean,
  accessibilityHaspopup?: boolean,
  accessibilityLabel?: string,
  color?: 'gray' | 'red' | 'blue' | 'transparent' | 'white',
  disabled?: boolean,
  inline?: boolean,
  name?: string,
  onClick?: ({ event: SyntheticMouseEvent<> }) => void,
  size?: 'sm' | 'md' | 'lg',
  text: string,
  type?: 'submit' | 'button',
|};

export default function Button(props: Props) {
  const {
    accessibilityExpanded,
    accessibilityHaspopup,
    accessibilityLabel,
    color = 'gray',
    disabled = false,
    inline = false,
    name,
    onClick,
    size = 'md',
    text,
    type = 'button',
  } = props;

  const textColor = {
    blue: 'white',
    gray: 'darkGray',
    red: 'white',
    transparent: 'white',
    white: 'darkGray',
  };

  const classes = classnames(styles.button, {
    [styles.sm]: size === 'sm',
    [styles.md]: size === 'md',
    [styles.lg]: size === 'lg',
    [styles.solid]: color !== 'transparent',
    [styles[color]]: !disabled,
    [styles.disabled]: disabled,
    [styles.enabled]: !disabled,
    [styles.inline]: inline,
    [styles.block]: !inline,
    [borderStyles.radiusLarge]: !inline,
    [borderStyles.radiusSmall]: inline,
  });

  /* eslint-disable react/button-has-type */
  return (
    <button
      aria-expanded={accessibilityExpanded}
      aria-haspopup={accessibilityHaspopup}
      aria-label={accessibilityLabel}
      className={classes}
      disabled={disabled}
      name={name}
      onClick={event => onClick && onClick({ event })}
      type={type}
    >
      <Text
        align="center"
        bold
        color={disabled ? 'gray' : textColor[color]}
        overflow="normal"
        size={size}
      >
        {text}
      </Text>
    </button>
  );
  /* eslint-enable react/button-has-type */
}

Button.propTypes = {
  accessibilityExpanded: PropTypes.bool,
  accessibilityHaspopup: PropTypes.bool,
  accessibilityLabel: PropTypes.string,
  color: PropTypes.oneOf(['blue', 'gray', 'red', 'transparent', 'white']),
  disabled: PropTypes.bool,
  inline: PropTypes.bool,
  name: PropTypes.string,
  onClick: PropTypes.func,
  size: PropTypes.oneOf(['sm', 'md', 'lg']),
  text: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['button', 'submit']),
};
