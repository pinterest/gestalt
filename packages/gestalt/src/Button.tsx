import { forwardRef, Fragment, ReactNode, useImperativeHandle, useRef } from 'react';
import classnames from 'classnames';
import NewTabAccessibilityLabel from './accessibility/NewTabAccessibilityLabel';
import styles from './Button.css';
import { useColorScheme } from './contexts/ColorSchemeProvider';
import Flex from './Flex';
import focusStyles from './Focus.css';
import Icon, { IconColor } from './Icon';
import icons from './icons/index';
import touchableStyles from './TapArea.css';
import Text from './Text';
import useFocusVisible from './useFocusVisible';
import useTapFeedback from './useTapFeedback';

const DEFAULT_TEXT_COLORS = {
  blue: 'inverse',
  gray: 'default',
  red: 'inverse',
  transparent: 'default',
  semiTransparentWhite: 'default',
  transparentWhiteText: 'inverse',
  white: 'default',
} as const;

const SIZE_NAME_TO_PIXEL = {
  sm: 10,
  md: 12,
  lg: 12,
} as const;

type Target = null | 'self' | 'blank';

type Props = {
  /**
   * Specifies the `id` of an associated element (or elements) whose contents or visibility are controlled by Button so that screen reader users can identify the relationship between elements. See the [Accessibility guidelines](https://gestalt.pinterest.systems/web/button#ARIA-attributes) for details on proper usage.
   */
  accessibilityControls?: string;
  /**
   * Indicates that Button hides or exposes collapsible components and expose whether they are currently expanded or collapsed. See the [Accessibility guidelines](https://gestalt.pinterest.systems/web/button#ARIA-attributes) for details on proper usage.
   */
  accessibilityExpanded?: boolean;
  /**
   * Indicates that a component controls the appearance of interactive popup elements, such as menu or dialog. See the [Accessibility guidelines](https://gestalt.pinterest.systems/web/button#ARIA-attributes) for details on proper usage.
   */
  accessibilityHaspopup?: boolean;
  /**
   * Label for screen readers to announce Button. See the [Accessibility guidelines](https://gestalt.pinterest.systems/web/button#ARIA-attributes) for details on proper usage.
   */
  accessibilityLabel?: string;
  /**
   * The background color of Button.
   * See the [color on white backgrounds variant](https://gestalt.pinterest.systems/web/button#Color-on-white-backgrounds) and the [color on color/image backgrounds variant](https://gestalt.pinterest.systems/web/button#Color-on-colorimage-backgrounds)
   */
  color?:
    | 'gray'
    | 'red'
    | 'blue'
    | 'transparent'
    | 'semiTransparentWhite'
    | 'transparentWhiteText'
    | 'white';
  /**
   * Available for testing purposes, if needed. Consider [better queries](https://testing-library.com/docs/queries/about/#priority) before using this prop.
   */
  dataTestId?: string;
  /**
   * Indicates if Button is disabled. Disabled Buttons are inactive and cannot be interacted with. See the [state variant](https://gestalt.pinterest.systems/web/button#State) for details on proper usage.
   */
  disabled?: boolean;
  /**
   * Default Buttons are sized by the text within the Button whereas full-width Buttons expand to the full width of their container. See the [width variant](https://gestalt.pinterest.systems/web/button#Size) variant to learn more.
   */
  fullWidth?: boolean;
  /**
   * An icon displayed after the text to help clarify the usage of Button. See the [icon variant](https://gestalt.pinterest.systems/web/button#Icons) to learn more.
   */
  iconEnd?: keyof typeof icons;
  /**
   * An icon displayed before the text to help clarify the Button usage.
   */
  iconStart?: keyof typeof icons;
  /**
   * The name attribute specifies the name of the button element. The name attribute is used to reference form-data after the form has been submitted and for [testing](https://testing-library.com/docs/queries/about/#priority).
   */
  name?: string;
  /**
   * Callback invoked when the user clicks (press and release) on Button with the mouse or keyboard.
   */
  onClick?: (arg1: {
    event: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLButtonElement>;
  }) => void;
  /**
   * Toggles between binary states: on/off, selected/unselected, open/closed. See the [selected](#Selected-state) variant to learn more. See the [state variant](https://gestalt.pinterest.systems/web/button#State) for details on proper usage.
   */
  selected?: boolean;
  /**
   * sm: 32px, md: 40px, lg: 48px
   *
   * See the [size variant](https://gestalt.pinterest.systems/web/button#Size) variant to learn more.
   */
  size?: 'sm' | 'md' | 'lg';
  /**
   * Use "-1" to remove Button from keyboard navigation. See the [Accessibility guidelines](https://gestalt.pinterest.systems/foundations/accessibility) to learn more.
   */
  tabIndex?: -1 | 0;
  /**
   * Text to render inside the Button to convey the function and purpose of the Button.
   */
  text: string;
  /**
   * Use "submit" if Button is used within or associated with a form.
   */
  type?: 'button' | 'submit';
};

function InternalButtonContent({
  target,
  text,
  textColor,
  iconStart,
  iconEnd,
  size,
}: {
  target?: Target;
  text: ReactNode;
  textColor: IconColor;
  iconStart?: keyof typeof icons;
  iconEnd?: keyof typeof icons;
  size: string;
}) {
  return (
    <Fragment>
      <Flex alignItems="center" gap={{ row: 2, column: 0 }} justifyContent="center">
        {iconStart && (
          <Icon
            accessibilityLabel=""
            color={textColor as IconColor}
            icon={iconStart}
            // @ts-expect-error - TS7053 - Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ readonly sm: 10; readonly md: 12; readonly lg: 12; }'.
            size={SIZE_NAME_TO_PIXEL[size]}
          />
        )}
        {text}
        {iconEnd ? (
          <Icon
            accessibilityLabel=""
            color={textColor}
            icon={iconEnd}
            // @ts-expect-error - TS7053 - Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ readonly sm: 10; readonly md: 12; readonly lg: 12; }'.
            size={SIZE_NAME_TO_PIXEL[size]}
          />
        ) : null}
      </Flex>
      <NewTabAccessibilityLabel target={target} />
    </Fragment>
  );
}

/**
 * [Buttons](https://gestalt.pinterest.systems/web/button) allow users to perform actions within a surface. They can be used alone for immediate action, or as a trigger for another component, like [Dropdown](https://gestalt.pinterest.systems/web/dropdown) or [Popover](https://gestalt.pinterest.systems/web/popover).
 *
 * ![Button light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Button.spec.ts-snapshots/Button-chromium-darwin.png)
 * ![Button dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Button-dark.spec.ts-snapshots/Button-dark-chromium-darwin.png)
 *
 */

const ButtonWithForwardRef = forwardRef<HTMLButtonElement, Props>(function Button(
  {
    accessibilityLabel,
    color = 'gray',
    dataTestId,
    disabled = false,
    fullWidth = false,
    iconEnd,
    iconStart,
    onClick,
    tabIndex = 0,
    selected = false,
    size = 'md',
    text,
    type,
    name,
    accessibilityControls,
    accessibilityExpanded,
    accessibilityHaspopup,
  }: Props,
  ref,
) {
  const innerRef = useRef<null | HTMLButtonElement>(null);

  // When using both forwardRef and innerRef, React.useimperativehandle() allows a parent component
  // that renders <Button ref={inputRef} /> to call inputRef.current.focus()
  // @ts-expect-error - TS2322 - Type 'HTMLButtonElement | null' is not assignable to type 'HTMLButtonElement'.
  useImperativeHandle(ref, () => innerRef.current);

  const {
    compressStyle,
    isTapping,
    handleBlur,
    handleMouseDown,
    handleMouseUp,
    handleTouchStart,
    handleTouchMove,
    handleTouchCancel,
    handleTouchEnd,
  } = useTapFeedback({
    height: innerRef?.current?.clientHeight,
    width: innerRef?.current?.clientWidth,
  });

  const { colorSchemeName } = useColorScheme();
  // We need to make a few exceptions for accessibility reasons in darkMode for red buttons
  const isDarkMode = colorSchemeName === 'darkMode';
  const isDarkModeRed = isDarkMode && color === 'red';

  const colorClass = color === 'transparentWhiteText' ? 'transparent' : color;

  const { isFocusVisible } = useFocusVisible();

  const sharedTypeClasses = classnames(styles.button, {
    [styles.inline]: !fullWidth,
    [styles.block]: fullWidth,
    [focusStyles.hideOutline]: !disabled && !isFocusVisible,
    [focusStyles.accessibilityOutline]: !disabled && isFocusVisible,
  });

  const baseTypeClasses = classnames(sharedTypeClasses, touchableStyles.tapTransition, {
    [styles.sm]: size === 'sm',
    [styles.md]: size === 'md',
    [styles.lg]: size === 'lg',
    [styles[colorClass]]: !disabled && !selected,
    [styles.selected]: !disabled && selected,
    [styles.disabled]: disabled,
    [styles.enabled]: !disabled,
    [touchableStyles.tapCompress]: !disabled && isTapping,
  });

  const parentButtonClasses = classnames(sharedTypeClasses, styles.parentButton);

  const childrenDivClasses = classnames(baseTypeClasses, styles.childrenDiv);

  const textColor =
    (disabled && 'disabled') ||
    (selected && 'inverse') ||
    (isDarkModeRed && 'default') ||
    DEFAULT_TEXT_COLORS[color];

  const buttonText = (
    <Text
      align="center"
      color={textColor}
      overflow="normal"
      size={size === 'sm' ? '200' : '300'}
      weight="bold"
    >
      {text}
    </Text>
  );

  if (type === 'submit') {
    return (
      <button
        ref={innerRef}
        aria-label={accessibilityLabel}
        className={baseTypeClasses}
        data-test-id={dataTestId}
        disabled={disabled}
        name={name}
        onBlur={handleBlur}
        onClick={(event) => onClick?.({ event })}
        onMouseDown={handleMouseDown}
        onMouseUp={handleMouseUp}
        onTouchCancel={handleTouchCancel}
        onTouchEnd={handleTouchEnd}
        // @ts-expect-error - TS2322 - Type '(arg1: TouchEvent<HTMLDivElement>) => void' is not assignable to type 'TouchEventHandler<HTMLButtonElement>'.
        onTouchMove={handleTouchMove}
        // @ts-expect-error - TS2322 - Type '(arg1: TouchEvent<HTMLDivElement>) => void' is not assignable to type 'TouchEventHandler<HTMLButtonElement>'.
        onTouchStart={handleTouchStart}
        style={compressStyle || undefined}
        // @ts-expect-error - TS2322 - Type '0 | -1 | null' is not assignable to type 'number | undefined'.
        tabIndex={disabled ? null : tabIndex}
        type="submit"
      >
        <InternalButtonContent
          iconEnd={iconEnd}
          size={size}
          text={buttonText}
          textColor={textColor}
        />
      </button>
    );
  }

  return (
    <button
      ref={innerRef}
      aria-controls={accessibilityControls}
      aria-expanded={accessibilityExpanded}
      aria-haspopup={accessibilityHaspopup}
      aria-label={accessibilityLabel}
      className={parentButtonClasses}
      data-test-id={dataTestId}
      disabled={disabled}
      name={name}
      onBlur={handleBlur}
      onClick={(event) => onClick?.({ event })}
      onMouseDown={handleMouseDown}
      onMouseUp={handleMouseUp}
      onTouchCancel={handleTouchCancel}
      onTouchEnd={handleTouchEnd}
      // @ts-expect-error - TS2322 - Type '(arg1: TouchEvent<HTMLDivElement>) => void' is not assignable to type 'TouchEventHandler<HTMLButtonElement>'.
      onTouchMove={handleTouchMove}
      // @ts-expect-error - TS2322 - Type '(arg1: TouchEvent<HTMLDivElement>) => void' is not assignable to type 'TouchEventHandler<HTMLButtonElement>'.
      onTouchStart={handleTouchStart}
      // @ts-expect-error - TS2322 - Type '0 | -1 | null' is not assignable to type 'number | undefined'.
      tabIndex={disabled ? null : tabIndex}
      type="button"
    >
      <div className={childrenDivClasses} style={compressStyle || undefined}>
        {iconEnd || iconStart ? (
          <InternalButtonContent
            iconEnd={iconEnd}
            iconStart={iconStart}
            size={size}
            text={buttonText}
            textColor={textColor}
          />
        ) : (
          buttonText
        )}
      </div>
    </button>
  );
});

ButtonWithForwardRef.displayName = 'Button';

export default ButtonWithForwardRef;
