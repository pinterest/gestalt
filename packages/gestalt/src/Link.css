.link {
  composes: noUnderline from "./Typography.css";
  color: inherit; /* Links should have the correct underline color */
  outline: none;
}

.link:hover,
.link:focus {
  text-decoration: underline;
}

.accessibleFocusStyle {
  composes: accessibilityOutline from "./Focus.css";
}

.block {
  composes: block from "./Layout.css";
}
