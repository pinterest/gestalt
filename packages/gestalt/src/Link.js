// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './Link.css';

type Props = {|
  children?: React.Node,
  href: string,
  inline?: boolean,
  onClick?: ({ event: SyntheticMouseEvent<> }) => void,
  target?: null | 'self' | 'blank',
|};

type State = {|
  enableFocusStyles: boolean,
|};

const TAB_KEY_CODE = 9;

export default class Link extends React.Component<Props, State> {
  static propTypes = {
    children: PropTypes.node.isRequired,
    href: PropTypes.string.isRequired,
    inline: PropTypes.bool,
    onClick: PropTypes.func,
    target: PropTypes.oneOf([null, 'self', 'blank']),
  };

  state: State = {
    enableFocusStyles: true,
  };

  handleClick = (event: SyntheticMouseEvent<>) => {
    const { href, onClick } = this.props;
    if (onClick && href) {
      onClick({ event });
    }
  };

  handleMouseDown = () => {
    const { href, target } = this.props;
    if (target === 'blank' && href) {
      this.setState({ enableFocusStyles: false });
    }
  };

  handleKeyUp = (event: SyntheticKeyboardEvent<>) => {
    const { href, target } = this.props;
    if (target === 'blank' && event.keyCode === TAB_KEY_CODE && href) {
      this.setState({ enableFocusStyles: true });
    }
  };

  render() {
    const { children, inline = false, target = null, href } = this.props;
    const rel = target === 'blank' ? 'noopener noreferrer' : null;
    const linkTarget = target ? `_${target}` : null;

    return (
      <a
        className={cx(
          styles.link,
          this.state.enableFocusStyles ? styles.accessibleFocusStyle : '',
          inline ? '' : styles.block
        )}
        href={href}
        onMouseDown={this.handleMouseDown}
        onKeyUp={this.handleKeyUp}
        onClick={this.handleClick}
        rel={rel}
        target={linkTarget}
      >
        {children}
      </a>
    );
  }
}
