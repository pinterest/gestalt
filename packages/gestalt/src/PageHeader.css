.pageHeader {
  width: 100%;
}

.pageHeaderBorderBottom {
  border-bottom: 1px solid var(--color-border-container);
  width: 100%;
}
