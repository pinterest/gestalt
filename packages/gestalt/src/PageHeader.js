// @flow strict
import React, { type Node, type Element } from 'react';
import Box from './Box.js';
import Button from './Button.js';
import IconButton from './IconButton.js';
import Link from './Link.js';
import Tooltip from './Tooltip.js';
import Flex from './Flex.js';
import Heading from './Heading.js';
import Text from './Text.js';
import { type Dimension } from './boxTypes.js';

type Props = {|
  title: string,
  maxWidth?: Dimension,
  primaryAction?: Element<typeof Button | typeof IconButton | typeof Link | typeof Tooltip>,
  secondaryAction?: Element<typeof Button | typeof IconButton | typeof Link | typeof Tooltip>,
  subtext?: string,
|};

export default function PageHeader({
  maxWidth = '100%',
  primaryAction,
  secondaryAction,
  subtext,
  title,
}: Props): Node {
  return (
    <Box color="white" paddingX={8} width="100%">
      <Flex flex="grow" justifyContent="center" maxWidth="100%">
        <Flex alignItems="center" flex="grow" justifyContent="between" maxWidth={maxWidth}>
          <Box marginEnd={4} minWidth={0} marginTop={2} marginBottom={2}>
            <Heading size="md" truncate accessibilityLevel={1}>
              {title}
            </Heading>
            {subtext && (
              <Box marginTop={2}>
                <Text truncate>{subtext}</Text>
              </Box>
            )}
          </Box>
          {primaryAction && (
            <Flex alignItems="center" gap={2} flex="none">
              {secondaryAction}
              {/* 48px height needed to maintain proper sizing when action is a Link */}
              <Box alignItems="center" display="flex" marginTop={4} marginBottom={4} height="48px">
                {primaryAction}
              </Box>
            </Flex>
          )}
        </Flex>
      </Flex>
    </Box>
  );
}
