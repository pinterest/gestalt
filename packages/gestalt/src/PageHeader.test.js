// @flow strict
import React from 'react';
import { create } from 'react-test-renderer';
import PageHeader from './PageHeader.js';
import Button from './Button.js';
import Link from './Link.js';

describe('PageHeader', () => {
  it('renders', () => {
    const tree = create(<PageHeader title="Settings" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders subtext', () => {
    const tree = create(<PageHeader title="Settings" subtext="5 followers" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders within a max width', () => {
    const tree = create(
      <PageHeader title="Settings" subtext="5 followers" maxWidth="60%" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders primary action', () => {
    const tree = create(
      <PageHeader
        title="Settings"
        primaryAction={<Button size="lg" color="red" text="Create" />}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders secondary action', () => {
    const tree = create(
      <PageHeader
        title="Settings"
        primaryAction={<Button size="lg" color="red" text="Create" />}
        secondaryAction={<Link href="www.pinterest.com">Help center</Link>}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
