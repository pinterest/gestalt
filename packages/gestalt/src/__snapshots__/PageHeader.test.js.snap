// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders 1`] = `
<div
  className="box paddingX8 whiteBg"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap0 flexGrow justifyCenter xsDirectionRow"
    style={
      Object {
        "maxWidth": "100%",
      }
    }
  >
    <div
      className="Flex rowGap0 flexGrow itemsCenter justifyBetween xsDirectionRow"
      style={
        Object {
          "maxWidth": "100%",
        }
      }
    >
      <div
        className="box marginBottom2 marginEnd4 marginTop2"
        style={
          Object {
            "minWidth": 0,
          }
        }
      >
        <h1
          className="Heading fontSize2 darkGray alignLeft breakWord truncate"
          title="Settings"
        >
          Settings
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageHeader renders primary action 1`] = `
<div
  className="box paddingX8 whiteBg"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap0 flexGrow justifyCenter xsDirectionRow"
    style={
      Object {
        "maxWidth": "100%",
      }
    }
  >
    <div
      className="Flex rowGap0 flexGrow itemsCenter justifyBetween xsDirectionRow"
      style={
        Object {
          "maxWidth": "100%",
        }
      }
    >
      <div
        className="box marginBottom2 marginEnd4 marginTop2"
        style={
          Object {
            "minWidth": 0,
          }
        }
      >
        <h1
          className="Heading fontSize2 darkGray alignLeft breakWord truncate"
          title="Settings"
        >
          Settings
        </h1>
      </div>
      <div
        className="Flex rowGap2 flexNone itemsCenter xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <div
            className="box itemsCenter marginBottom4 marginTop4 xsDisplayFlex"
            style={
              Object {
                "height": "48px",
              }
            }
          >
            <button
              className="button hideOutline tapTransition lg red enabled block accessibilityOutline"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <div
                className="Text fontSize3 white alignCenter fontWeightBold"
              >
                Create
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageHeader renders secondary action 1`] = `
<div
  className="box paddingX8 whiteBg"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap0 flexGrow justifyCenter xsDirectionRow"
    style={
      Object {
        "maxWidth": "100%",
      }
    }
  >
    <div
      className="Flex rowGap0 flexGrow itemsCenter justifyBetween xsDirectionRow"
      style={
        Object {
          "maxWidth": "100%",
        }
      }
    >
      <div
        className="box marginBottom2 marginEnd4 marginTop2"
        style={
          Object {
            "minWidth": 0,
          }
        }
      >
        <h1
          className="Heading fontSize2 darkGray alignLeft breakWord truncate"
          title="Settings"
        >
          Settings
        </h1>
      </div>
      <div
        className="Flex rowGap2 flexNone itemsCenter xsDirectionRow"
      >
        <div
          className="FlexItem"
        >
          <a
            className="link hideOutline tapTransition block rounding0 hoverUnderline accessibilityOutline"
            href="www.pinterest.com"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            rel=""
            target={null}
          >
            Help center
          </a>
        </div>
        <div
          className="FlexItem"
        >
          <div
            className="box itemsCenter marginBottom4 marginTop4 xsDisplayFlex"
            style={
              Object {
                "height": "48px",
              }
            }
          >
            <button
              className="button hideOutline tapTransition lg red enabled block accessibilityOutline"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <div
                className="Text fontSize3 white alignCenter fontWeightBold"
              >
                Create
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageHeader renders subtext 1`] = `
<div
  className="box paddingX8 whiteBg"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap0 flexGrow justifyCenter xsDirectionRow"
    style={
      Object {
        "maxWidth": "100%",
      }
    }
  >
    <div
      className="Flex rowGap0 flexGrow itemsCenter justifyBetween xsDirectionRow"
      style={
        Object {
          "maxWidth": "100%",
        }
      }
    >
      <div
        className="box marginBottom2 marginEnd4 marginTop2"
        style={
          Object {
            "minWidth": 0,
          }
        }
      >
        <h1
          className="Heading fontSize2 darkGray alignLeft breakWord truncate"
          title="Settings"
        >
          Settings
        </h1>
        <div
          className="box marginTop2"
        >
          <div
            className="Text fontSize3 darkGray alignLeft breakWord fontWeightNormal truncate"
            title="5 followers"
          >
            5 followers
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageHeader renders within a max width 1`] = `
<div
  className="box paddingX8 whiteBg"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="Flex rowGap0 flexGrow justifyCenter xsDirectionRow"
    style={
      Object {
        "maxWidth": "100%",
      }
    }
  >
    <div
      className="Flex rowGap0 flexGrow itemsCenter justifyBetween xsDirectionRow"
      style={
        Object {
          "maxWidth": "60%",
        }
      }
    >
      <div
        className="box marginBottom2 marginEnd4 marginTop2"
        style={
          Object {
            "minWidth": 0,
          }
        }
      >
        <h1
          className="Heading fontSize2 darkGray alignLeft breakWord truncate"
          title="Settings"
        >
          Settings
        </h1>
        <div
          className="box marginTop2"
        >
          <div
            className="Text fontSize3 darkGray alignLeft breakWord fontWeightNormal truncate"
            title="5 followers"
          >
            5 followers
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
